### PR TITLE
Preparing new HTML structure

### PR DESCRIPTION
**Pull request by jbidoret**
In CSS, introduction of first variables, layout helpers, basic styles for inputs, buttons and fields.
In HTML, first step of the restructuration: main wrappers for top bar and nav bar
With `layout.php`, it mainly consists of re-indentation, light reorganization and the setup of a new strategy to solve #444.

### CODE REVIEW

File: app/view/templates/layout.php [hunk 1]
```diff
@@ -23,37 +25,31 @@
...
-                <li class="alert alert-<?= $flashmessage['type'] ?>">
-                    <?= $flashmessage['content'] ?>
-                </li>
-            <?php endforeach ?>
-            </ul>
-        </div>
-    </a>
-<?php endif ?>
-
 <body>
```
Comment (vincent-peugnet) [inline app/view/templates/layout.php:36]: Oh! interesting, I totally forgot that I've put flash messages out of body. Seems better like this !

**Comment by vincent-peugnet** [comment]
A quick comment: I prefer to keep the JS/no-JS detection for another PR. For me commit 33a1804f54fc678da68c0c9b02f54575e89aa7b3 could be out of this one.

**Comment by jbidoret** [comment]
> A quick comment: I prefer to keep the JS/no-JS detection for another PR. For me commit [33a1804](https://github.com/vincent-peugnet/wcms/commit/33a1804f54fc678da68c0c9b02f54575e89aa7b3) could be out of this one.

Don’t know how to remove the commit from the PR…

File: assets/css/media.css [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+    width: 100%;
+    border-radius: 0 0 var(--radius) var(--radius);
+}
+.dropped-files-info span:has(+ span)::after { 
+    content: ", ";
+}
+#file {width: 100%;}
+#file:has(+.dropped-files-info){
+    border-radius: var(--radius) var(--radius) 0 0;
+}
```
Comment (vincent-peugnet) [inline assets/css/media.css:16]: For example, I would have keep all this for another PR since this is enhancement.

**Comment by jbidoret** [comment]
> Don’t know how to remove the commit from the PR…
Ok. Simple :)

File: assets/css/media.css (hunk 2 above)
Comment (jbidoret) [reply assets/css/media.css:16]: I understand what you mean.
This CSS snippet should have been included in  the PR relative to the drag and drop feature.
But while cleaning the File dropdown in media, it would have been weird to me to leave unstyled the drag-n-drop info panel.

Comment (jbidoret) [reply assets/css/media.css]: I removed the commit from the PR

Comment (vincent-peugnet) [reply assets/css/media.css]: > it would have been weird to me to leave unstyled the drag-n-drop info panel.

😁 haha yes but, from how I understand it, that's exactly the point of the commit narration: One commit should tell only one story.

**Comment by vincent-peugnet** [comment]
Thanks for the work Julien !!!

Using the default theme, almost everything seems fine to me.

But I see that themes that use borders, like `dark-doriphore.css` and `industrial-dream.css` have some issues since they're not updated according to new HTML.

**Comment by vincent-peugnet** [comment]
One of the thing which is different from the old UI is the "selected menu". Now the font weight get bigger as before it was underlined. I think that this PR should keep the old style to let room for choosing a new style in a different PR. Cause currently I see two issues:

- when menu item from the right side is selected, this may change the item's width and move a little bit all the other.
- on mobile, current menu is not highlighted, which is a regression.

**Comment by vincent-peugnet** [comment]
Ok, I'm checking the mobile and many little things seems to be broken.

Here is a quick list of thing that should behave like before:

- dropdowns of the `nav.hbar` (should be aligned to the left)
- `form#search` shoud disapear
- strange behavior of edit/view/delete/download buttons

Do you need help on this @jbidoret ?

**Comment by vincent-peugnet** [comment]
this commentary:
```
/* --------------------------------------------------------- almost untouched… */
```
have some meaning at the time of this PR, but in the future I'm afraid that it could be misleading 😁

**Comment by vincent-peugnet** [comment]
If you prefer the style of separator commentary you've used, let change them all in order to unify things !

Let's upgrade thoose `/* __________________________ Flash Messages ___________________________ */`

**Comment by vincent-peugnet** [comment]
And if you still have energy, it would be awesome to de a little check to see if there are some dead CSS code.
